bin/mycroft-stop

```diff
@@ -14,8 +14,107 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-SOURCE="${BASH_SOURCE[0]}"
-cd -P "$( dirname "$SOURCE" )"/.. || exit
-DIR="$( pwd )"
+script=${0}
+script=${script##*/}
 
-. "$DIR/stop-mycroft.sh" "$@"
+help() {
+	echo "${script}:  Mycroft service stopper"
+	echo "usage: ${script} [service]"
+	echo
+	echo "Service:"
+	echo "  all       ends core services: bus, audio, skills, voice"
+	echo "  (none)    same as \"all\""
+	echo "  bus       stop the Mycroft messagebus service"
+	echo "  audio     stop the audio playback service"
+	echo "  skills    stop the skill service"
+	echo "  voice     stop voice capture service"
+	echo "  enclosure stop enclosure (hardware/gui interface) service"
+	echo
+	echo "Examples:"
+	echo "  ${script}"
+	echo "  ${script} audio"
+
+	exit 0
+}
+
+process_running() {
+	if [ "$( pgrep -f "python3 (.*)-m mycroft.*${1}" )" ]; then
+		return 0
+	else
+		return 1
+	fi
+}
+
+end_process() {
+	if process_running "$1"; then
+		# Find the process by name, only returning the oldest if it has children
+		pid=$( pgrep -o -f "python3 (.*)-m mycroft.*${1}" )
+		echo "Stopping $1 (${pid})..."
+		kill -s SIGINT "${pid}"
+
+		# Wait up to 5 seconds (50 * 0.1) for process to stop
+		c=1
+		while [ $c -le 50 ]; do
+			if process_running "$1"; then
+				sleep 0.1
+				c=$((c + 1))
+			else
+				c=999 # end loop
+			fi
+		done
+
+		if process_running "$1"; then
+			echo "Failed to stop."
+			pid=$( pgrep -o -f "python3 (.*)-m mycroft.*${1}" )
+			echo "  Killing $1 (${pid})..."
+			kill -9 "${pid}"
+			echo "Killed."
+			result=120
+		else
+			echo "Stopped."
+			if [ $result -eq 0 ] ; then
+				result=100
+			fi
+		fi
+	fi
+}
+
+result=0 # default, no change
+
+OPT=$1
+shift
```

Comment:
This gives error if there are no arguments, maybe add a check similar to [stop-mycroft.sh L91](https://github.com/MycroftAI/mycroft-core/blob/dev/stop-mycroft.sh#L91)